### ISSUE TITLE
[mobx] MobX 5+ requires Proxy objects.

### ISSUE THREAD

**Issue by cagataycali** [mobxjs/mobx#1634]
- [x] Issue:
  * [x] Can't find variable: Symbol (I fixed with `import 'es6-symbol/implement'`)
  * [ ] **MobX 5+ requires Proxy objects. If your environment doesn't support Proxy objects, please downgrade to MobX 4.**

I'm trying to run MobX as a state manager in little react-native project. It works on ios but fails in android device.

Here's the store.js

```javascript
import { AsyncStorage } from 'react-native'
import { observable, computed, action } from 'mobx'
import { persist } from 'mobx-persist'

class Store {
  @persist('object') @observable user = {}
  @persist @observable token = null

  @computed get loggedIn () {
    return this.user && this.token !== null
  }

  login (argv, token) {
    this.user = argv
    this.token = token
    console.log('User logged in')
  }

  async getContent () {
    return fetch('https://cagatay.pro/small-image.json').then(response => response.json())
  }

  logout () {
    this.user = null
    this.token = null
  }
}

export default new Store()
```

**Comment by jamiewinder** [comment]
Yes, MobX 5+ does require proxies. Version 5 was the big proxy-based rewrite.

If you need to use MobX in an environment that doesn't support proxies, you need to stick with MobX 4 (as I do, relatively painlessly).

**Comment by badarshahzad** [comment]
Facing the same error :disappointed: can anyone tell me how to resolve this one?

**Comment by badarshahzad** [comment]
You easily closed the issue? Do you know I have still the same error

**Comment by mattruby** [comment]
It's clearly the same issue. Downgrade mobx to 4.

If your issue persists, open a new issue.

**Comment by pincheira** [comment]
Solution is to update JavaScriptCore on your RN Android project as @mweststrate's suggested: https://github.com/react-community/jsc-android-buildscripts#how-to-use-it-with-my-react-native-app

**Comment by bluepeter** [comment]
This, uh, means that MobX is not viable on Internet Explorer 11...

**Comment by mweststrate** [comment]
See top the readme....

Op wo 29 aug. 2018 20:11 schreef bluepeter <notifications@github.com>:

**Comment by mweststrate** [comment]
Locking this issue. All issues reported so far here can be solved by just reading the readme. It clearly states which version is supported on which platform and that both 4.x and 5.x are actively maintained.
